mlir/include/mlir/Support/Timing.h

```diff
@@ -473,6 +473,13 @@ void registerDefaultTimingManagerCLOptions();
 /// 'registerDefaultTimingManagerOptions' to a `DefaultTimingManager`.
 void applyDefaultTimingManagerCLOptions(DefaultTimingManager &tm);
 
+/// Return a default output strategy for the specified format. This function can
+/// be used in combination with DefaultTimingManager::setOutput() to use
+/// MLIR-provided output format.
+std::unique_ptr<OutputStrategy>
+createDefaultOutputStrategy(DefaultTimingManager::OutputFormat fmt,
+                            raw_ostream &os);
```

Comment:
```suggestion
/// Create an output strategy for the specified format, to be passed to
/// DefaultTimingManager::setOutput().
std::unique_ptr<OutputStrategy>
createOutputStrategy(DefaultTimingManager::OutputFormat fmt,
                            raw_ostream &os);
```

I don't know why "default" needs to be in the name here?

We have another issue here which is one of namespace: most names in this header are too generic to be in the `mlir` namespace, we should move all this into a `mlir::timing` namespace instead. If you're up to it feel free to send a follow-up PR.

Reply:
went with "default" since users can implement their own output strategy from scratch, MLIR just gives "basic" implementations for text and json. but fair enough, no reason to explicitly say so, i guess it's clear from the API.